src/main/java/io/logz/log4j2/LogzioAppender.java

```diff
@@ -263,37 +295,80 @@ public void start() {
         } catch (UnknownHostException e) {
             statusLogger.warn("The configuration addHostName was specified but the host could not be resolved, thus the field 'hostname' will not be added", e);
         }
-        String bufferDirPath;
-        if (bufferDir != null) {
-            bufferDirPath = bufferDir;
-            File bufferFile = new File(bufferDirPath);
-            if (bufferFile.exists()) {
-                if (!bufferFile.canWrite()) {
-                    statusLogger.error("We cant write to your bufferDir location: "+bufferFile.getAbsolutePath());
+
+        if (!inMemoryBuffer) {
```

Comment:
Use this flag, in the middle of the builder, only for the section you're building the queue, thus save boilerplate code.